### PR TITLE
Add go-eigentrust protobuf definition

### PR DESCRIPTION
**Pull request by astralblue**
Served by go-eigentrust.  Can be used to:

* Manage local trust matrix, pre-trust vector, and initial/result trust vector (`(Create|Flush|Delete)Trust(Matrix|Vector)`)
* Push updates to the local trust and pre-trust (`UpdateTrust(Matrix|Vector)`)
* Trigger one-shot EigenTrust compute (`BasicCompute`)
* Pull the compute result (`GetTrustVector`)
* Pull the local trust snapshot (`GetTrustMatrix`) for debugging/inspection

Also includes push-based result uploading mechanics, currently unimplemented.

### CODE REVIEW

**Comment by astralblue** [comment]
Targeting the main branch, requires nothing from indexer or column family branch

**Comment by lazovicff** [comment]
@astralblue Looks good, but still needs to pass formatting to get merged

File: proto-buf/services/eigentrust.proto [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+message TrustMatrixHeader {
+  optional string id = 1;
+  repeated uint64 timestamp_qwords = 2;
+}
+
+message TrustMatrixEntry {
+  string truster = 1;
+  string trustee = 2;
+  double value = 3;
+}
```
Comment (lazovicff) [inline proto-buf/services/eigentrust.proto:29]: This should also include timestamp, no?

Comment (astralblue) [reply proto-buf/services/eigentrust.proto:29]: No, the timestamp is in the header.  The stream contains one header part, followed by zero or more entry parts.  All the entries share the same timestamp in the header preceding them.

Comment (astralblue) [reply proto-buf/services/eigentrust.proto:29]: Roger, fixing.

**Comment by astralblue** [comment]
> @astralblue Looks good, but still needs to pass formatting to get merged

Roger, fixing

File: proto-buf/services/eigentrust.proto [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  // Update a trust matrix.
+  // Entries for the same matrix with the same timestamp can be batched
+  // in the same request.
+  // Updates cannot go back in time:
+  // An update older than the target matrix is rejected.
+  // Updates can be empty (have zero entries):
+  // This can be used to force-update the timestamp of the trust matrix
+  // and trigger periodic re-compute of a compute job.
+  rpc UpdateTrustMatrix(UpdateTrustMatrixRequest)
+      returns (UpdateTrustMatrixResponse) {}
```
Comment (lazovicff) [inline proto-buf/services/eigentrust.proto]: So I will have to use this to push LT matrix items into the core compute?
Is there a logic inside CoreCompute to pull the terms from LC component?

Comment (astralblue) [reply proto-buf/services/eigentrust.proto]: Yes, that pulling was what I've doing so far.  Pushing LT entries as they are updated streamlines processing more evenly though, and is the way to go, so I would ask LC to push terms.  (Don't want to keep the pulling logic much longer if possible at all.)  P1 though.

Comment (lazovicff) [reply proto-buf/services/eigentrust.proto]: So I should think about how to use this to push terms to CoreCompute for this milestone as well?

BTW, what does P1 mean?

Comment (astralblue) [reply proto-buf/services/eigentrust.proto]: No, not for this milestone.  I can continue using the puller.  That's what I meant by P1 (priority 1), as opposed to P0 (priority 0, i.e. must be done this milestone). :-)

Comment (astralblue) [reply proto-buf/services/eigentrust.proto]: I don't want to have the pulling logic in the long-running system such as Milestone 2. :-p

**Review by lazovicff — APPROVED** [review]

**Pull request merged**